mindmeld/active_learning/classifiers.py

```diff
@@ -531,3 +678,120 @@ def _update_eval_stats_intent_level(
                         self.class_level_statistic
                     ][i]
                 }
+
+    def entity_recognizers_fit_eval(
+        self,
+        sampled_queries: ProcessedQueryList,
+        unsampled_queries: ProcessedQueryList,
+        test_queries: ProcessedQueryList,
+        domain_to_intents: Dict,
+        entity2id: Dict,
+    ):
+        """Fit and evaluate the entity recognizer.
+        Args:
+            sampled_queries (ProcessedQueryList): List of Sampled Queries.
+            unsampled_queries (ProcessedQueryList): List of Unsampled Queries.
+            test_queries (ProcessedQueryList): List of Test Queries.
+            domain_to_intents (Dict): Dictionary mapping domain to list of intents.
+            entity2id (Dict): Dictionary mapping entities to IDs.
+
+        Returns:
+            ic_queries_prob_vectors (List[List]): List of probability distributions
+                for unsampled queries.
+            ic_eval_test_dict (Dict): Dictionary mapping a domain (str) to the
+                associated ic_eval_test object.
+        """
+        er_eval_test_dict = {}
+        unsampled_idx_preds_pairs = {}
+        for domain, intents in domain_to_intents.items():
+            for intent in intents:
+                # Filter Queries
+                (
```

Comment:
Nit: Is there a way to make these similar calls with a for loop to reduce some verbosity/repetition?

Reply:
It might become difficult to follow later, and the outputs are independent each time so variables will need to be added on regardless.